README.md

```diff
@@ -95,6 +95,136 @@ For ready to use installation, please refer to the README file at: https://githu
 Once server is started you can access a swagger via http://0.0.0.0:5050/swagger/ 
 
 
+### Endpoints
+
+##### List all projects
+```bash
+curl -X GET http://0.0.0.0:5050/projects/
+```
+
+##### Create a project
+```bash
+curl -X POST http://0.0.0.0:5050/projects/ \
+  -F file=@/path/to/your/video/SampleVideo.mp4
+```
+
+##### Retrieve project details
+```bash
+curl -X GET http://0.0.0.0:5050/projects/5d7b841764c598157d53ef4a
+```
+where `5d7b841764c598157d53ef4a` is project's `_id`.
+
+##### Delete a project
+```bash
+curl -X DELETE http://0.0.0.0:5050/projects/5d7b841764c598157d53ef4a
+```
+where `5d7b841764c598157d53ef4a` is project's `_id`.
+
+
+##### Duplicate a project
+```bash
+curl -X POST http://0.0.0.0:5050/projects/5d7b841764c598157d53ef4a/duplicate
+```
+where `5d7b841764c598157d53ef4a` is project's `_id` you want to make a duplicate of.
+
+##### Edit
+
+:warning: It's not permitted to edit an original project (version 1), instead use a duplicated project.
+
+##### Trim
+```bash
+curl -X PUT \
+  http://0.0.0.0:5050/projects/5d7a35a04be797ba845e7871 \
+  -d '{
+	"trim": {
+		"start": 2,
+		"end": 5
+	}
+}'
+```
+where `2` and `5` are seconds.
+
+##### Rotate
+```bash
+curl -X PUT \
+  http://0.0.0.0:5050/projects/5d7a35a04be797ba845e7871 \
+  -d '{
+	"rotate": 90
+}'
+```
+where `90` is rotate degree.
+
+##### Scale
+```bash
+curl -X PUT \
+  http://0.0.0.0:5050/projects/5d7a35a04be797ba845e7871 \
+  -d '{
+	"scale": 480
+}'
+```
+where `480` is width you want to scale video to.
+
+##### Crop
+```bash
+curl -X PUT \
+  http://0.0.0.0:5050/projects/5d7a35a04be797ba845e7871 \
+  -d '{
+	"crop": {
+		"height": 180,
+		"width": 320,
+		"x": 0,
+		"y": 0
+	}
+}'
+```
+where `width` and `height` are respectively width and height of capturing area,
+and `x` and `y` are coordinates of top-left point of capturing area.
+https://ffmpeg.org/ffmpeg-filters.html#crop
+
+##### Capture timeline thumbnails
+```bash
+curl -X GET 'http://0.0.0.0:5050/projects/5d7b90ed64c598157d53ef5d/thumbnails?type=timeline&amount=5'
+```
+
+##### Capture a thumbnail for a preview at a certain position
+```bash
+curl -X GET 'http://0.0.0.0:5050/projects/5d7b98f52fac91d2e1ad7512/thumbnails?type=preview&position=5'
+```
+where `position` is a position in the video (seconds) used to capture a thumbnail. 
+
+You can also specify optional `crop` param if you want to crop a preview thumbnail, just add
+`crop={ "height": 180, "width": 320, "x": 0, "y": 0 }`.
```

Comment:
imo might be good to specify the crop with something like `crop=0,320,0,180` so it would be more consistent with other options

Reply:
Hey, I think in this case it's better to provide it like this: `crop=0,0,320,180` which is `x,y,width,height`. What do you think?

Reply:
yep makes sense. we can support json via request body but in url it's better like that imo